modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,212 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import {FieldBase} from 'frontend-js-components-web';
+import {fetch, navigate, objectToFormData} from 'frontend-js-web';
+import React, {useRef, useState} from 'react';
+
+type frontendTokenDefinitionProvider = {
+	name: string;
+	themeId: string;
+};
+
+interface AddStyleBookModalProps {
+	addStyleBookEntryURL: string;
+	closeModal: () => void;
+	frontendTokenDefinitionProviders?: Array<frontendTokenDefinitionProvider>;
+	namespace: string;
+}
+
+const AddStyleBookModalContent = ({
+	addStyleBookEntryURL,
+	closeModal,
+	frontendTokenDefinitionProviders = [],
+	namespace,
+}: AddStyleBookModalProps) =>{
+
+	const [active, setActive] = useState<boolean>(false);
+	const [loading, setLoading] = useState<boolean>(false);
+
+	const [errorMessage, setErrorMessage] = useState<string>();
+
+	const [name, setName] = useState<string>('');
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0]);
+
+	const formRef = useRef(null);
+
+	const handleFormError = (responseContent: any) => {
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const handleSubmit = (event: any) => {
+		event.preventDefault();
+
+		const error = name.trim().length
+			? ''
+			: Liferay.Language.get('this-field-is-required');
+
+		if (error) {
+			setErrorMessage(error);
+
+			return;
+		}
+
+		setLoading(true);
+
+		const body = Liferay.Util.ns(namespace, {
+			name,
+			themeId: themeId.themeId,
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrorMessage(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: closeModal,
+					});
+				}
+			})
+			.catch((response) =>
+				handleFormError(response)
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const themeIdId = `${namespace}tokenDefinition`;
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+
+				{errorMessage && (
+					<ClayAlert
+						displayType="danger"
+						onClose={() => {}}
+						title={Liferay.Language.get('error')}
+					>
+						{errorMessage}
+					</ClayAlert>
+				)}
+
+				<FieldBase
+					className="themeId"
+					helpMessage={Liferay.Language.get(
+						'the-style-book-will-be-created-based-on-the-selected-token-definition'
+					)}
+					id={themeIdId}
+					label={Liferay.Language.get('create-style-book-for')}
```

Comment:
For both inputs we should be using FieldBase's errorMessage prop to show the error message related to the field directly on the field.